### ISSUE TITLE
Automatically adding default fields to update 

### ISSUE THREAD

**Issue by RoyC-Retail-Assist** [linq2db/linq2db#862]
I have some audit type fields (UserName, UpdateDateTime etc) on certain tables that i would like to be automatically populated on Insert or Update (and the variations of them).  I have attributes on the properties for the fields that need to updated.

I have a partial solution based on the ConcurencyCheckTests sample which achieves what I need but the query is cached on the first call therefore doesn't update the fields for subsequent calls.

Do you have any suggestions as to a better way of doing this?  

```c#
        protected override SelectQuery ProcessQuery(SelectQuery selectQuery)
        {
            if (selectQuery.IsInsert || selectQuery.IsInsertOrUpdate || selectQuery.IsUpdate)
            {
                SqlTable source = null;
                if (selectQuery.IsInsert)
                {
                    source = selectQuery.Insert.Into;
                }
                else
                {
                    source = selectQuery.From.Tables[0].Source as SqlTable;
                }

                if (source != null)
                {
                    var descriptor = MappingSchema.GetEntityDescriptor(source.ObjectType);
                    if (descriptor != null)
                    {
                        var auditCols = descriptor.Columns.Select(c => new { col = c, attrib = c.MemberAccessor.GetAttribute<AuditField>() }).Where(c => c.attrib != null);

                        if (auditCols.Any())
                        {
                            var newQuery = Clone(selectQuery);

                            if (selectQuery.IsInsert)
                            {
                                foreach (var auditAttrib in auditCols)
                                {
                                    var field = newQuery.Insert.Into[auditAttrib.col.MemberName];

                                    var auditColumn = (from i in newQuery.Insert.Items
                                                       let f = i.Column as SqlField
                                                       where f != null && f.PhysicalName == field.PhysicalName
                                                       select i).FirstOrDefault();

                                    if (auditColumn != null)
                                    {
                                        // overwrite any data that has been provided so that we get the "correct" data recorded
                                        auditColumn.Expression = GetFieldValue(auditAttrib.attrib.FieldType);
                                    }
                                    else
                                    {
                                        // need to add in the audit column
                                        auditColumn = new SelectQuery.SetExpression(field, GetFieldValue(auditAttrib.attrib.FieldType));
                                        newQuery.Insert.Items.Add(auditColumn);
                                    }
                                }
                            }

                            return newQuery;
                        }
                    }
                }
            }

            return base.ProcessQuery(selectQuery);
        }
```

### Environment details
linq2db version: 1.9.0  
Database Server: Any
Database Provider: Any  
Operating system: Any
Framework version: .NET Framework 4.6.1

**Comment by sdanyliv** [comment]
Fixed via #863

**Comment by dark2201** [comment]
@sdanyliv 	what's the recommendation of handling this?

overriding `ProcessQuery` of `DataConnection`?

I want to automatically set `UpdatedDate` when a row is updated and `CreatedDate` when a row is inserted.
